### PR TITLE
Story4.5

### PR DESCRIPTION
**Pull request by kolya2po**

### CODE REVIEW

File: GameStore.DAL/RelationshipsConfiguration/CartConfiguration.cs [hunk 1]
```diff
@@ -11,6 +11,9 @@ public void Configure(EntityTypeBuilder<Cart> builder)
             builder.HasMany(c => c.CartItems)
                 .WithOne()
                 .HasForeignKey(c => c.CartId);
+
+            builder.Property(c => c.TotalPrice)
```
Comment (sivokon) [inline GameStore.DAL/RelationshipsConfiguration/CartConfiguration.cs:15]: how do we use this property?

File: GameStore.WebApi/Controllers/OrdersController.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Threading.Tasks;
+using AutoMapper;
+using GameStore.BLL.Interfaces;
+using GameStore.BLL.Models;
+using Microsoft.AspNetCore.Http;
```
Comment (sivokon) [inline GameStore.WebApi/Controllers/OrdersController.cs]: unused reference

File: GameStore.WebApi/Controllers/OrdersController.cs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            var cart = await _cartsService.GetByIdAsync(Convert.ToInt32(cartId));
+
+            if (cart == null)
+            {
+                return NotFound("Cart doesn't exist.");
+            }
+
+            if (Request.Cookies.TryGetValue("orderId", out var orderId))
+            {
+                var orderModel = await _ordersService.GetByIdAsync(Convert.ToInt32(orderId));
```
Comment (sivokon) [inline GameStore.WebApi/Controllers/OrdersController.cs]: Let's put this line inside the service, as part of update orderItems logic

File: GameStore.BLL/Services/OrderItemsService.cs [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        public async Task<IEnumerable<OrderItem>> GetAllByOrderIdAsync(int orderId)
+        {
+            return await UnitOfWork.OrderItemsRepository.GetAllByOrderIdAsync(orderId);
+        }
+
+        public async Task UpdateAsync(OrderModel orderModel, IEnumerable<CartItemModel> cartItems)
+        {
+            var itemsFromCart = Mapper.Map<IEnumerable<OrderItem>>(cartItems).ToArray();
+
+            FindAndDeleteObsoleteOrderItems(orderModel.OrderItems, itemsFromCart);
```
Comment (sivokon) [inline GameStore.BLL/Services/OrderItemsService.cs]: We can just delete all orderItems and create new ones. We will still have 2 calls to database - DeleteRange and CreateRange, so almost no difference from the performance perspective

File: GameStore.DAL/RelationshipsConfiguration/CartConfiguration.cs (hunk 1 above)
Comment (kolya2po) [reply GameStore.DAL/RelationshipsConfiguration/CartConfiguration.cs:15]: I've specified a concrete type bc I was asked to do so when I tried to update database.
Value of this property is automatically calculated.

**Review by sivokon — APPROVED** [review]

**Pull request merged**